internal/beater/otlp/map_caching.go

```diff
@@ -0,0 +1,71 @@
+package otlp
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/elastic/apm-server/internal/logs"
+	"github.com/elastic/apm-server/internal/sourcemap"
+	"github.com/elastic/elastic-agent-libs/logp"
+	lru "github.com/hashicorp/golang-lru"
+)
+
+type MapCachingFetcher struct {
+	cache   *lru.Cache
+	backend MapFetcher
+	logger  *logp.Logger
+}
+
+// NewMapCachingFetcher returns a MapCachingFetcher that wraps backend, caching results for the configured cacheExpiration.
+func NewMapCachingFetcher(
+	backend MapFetcher,
+	cacheSize int,
+) (*MapCachingFetcher, error) {
+	logger := logp.NewLogger(logs.Sourcemap)
+
+	lruCache, err := lru.NewWithEvict(cacheSize, func(key, value interface{}) {
+		logger.Debugf("Removed id %v", key)
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to create lru cache for map caching fetcher: %w", err)
+	}
+
+	return &MapCachingFetcher{
+		cache:   lruCache,
+		backend: backend,
+		logger:  logger,
+	}, nil
+}
+
+// Fetch fetches an android source map from the cache or wrapped backend.
+func (f *MapCachingFetcher) Fetch(ctx context.Context, name, version string) ([]byte, error) {
+	key := sourcemap.Identifier{
```

Comment:
I'd argue that we don't need to reuse anything from sourcemap package to avoid an unnecessary dependency on a mostly unrelated package. This package can have its own struct representing an identifier.

Reply:
Thanks, I've created an ide only for android maps caching.